react/week2/todo-part2/src/App.js

```diff
@@ -0,0 +1,128 @@
+import React, { useState, useEffect } from "react";
+import ToDos from "./ToDos";
+
+const Header = () => {
+  return <h1> Todo List</h1>;
+};
+
+const Timer = () => {
+  const [count, setCount] = useState(0);
+  useEffect(() => {
+    setTimeout(() => {
+      setCount((timer) => timer + 1);
+    }, 1000);
+  });
+  return <h3>You have used {count} seconds on this website</h3>;
+};
+// component to check the todo array when it is empty render a message
+const CheckTodoArray = (props) => {
+  if (props.todoArray.length === 0) {
+    return <div> No Items </div>;
+  } else {
+    return null;
+  }
+};
+// the main component for add and delete todo item
+const ListedTodo = () => {
+  const [todoArray, setToDoArray] = useState(ToDos);
+  console.log(todoArray);
+  //add todo function to execute modifying on todo array by useState
+  const addTodo = () => {
+    setToDoArray((prev) => {
+      return [
+        ...prev,
+        { id: Date.now(), description: "Random text", done: false },
+      ];
+    });
+  };
+
+  // delete function execute filter toDos array
+  const deleteTodo = (props) => {
+    setToDoArray((ToDos) => {
+      return ToDos.filter((todo) => {
+        return todo.id !== props.todoItems.id;
+      });
+    });
+  };
+  // add todo button component execute creat new todo item by button
+  const AddTodoButton = () => {
+    return (
+      <div>
+        <button onClick={addTodo}>Add Todo</button>
+        <CheckTodoArray todoArray={todoArray} />
+      </div>
+    );
+  };
```

Comment:
<Component /> should be created outside of your Parent component. Because otherwise it will keep "re-making/rendering" this every time you e.g. type/setValue onChange.

And pass the addTodo function to it